sway/desktop/xwayland.c

```diff
@@ -282,13 +282,19 @@ static void handle_commit(struct wl_listener *listener, void *data) {
 		wl_container_of(listener, xwayland_view, commit);
 	struct sway_view *view = &xwayland_view->view;
 	struct wlr_xwayland_surface *xsurface = view->wlr_xwayland_surface;
-	struct wlr_surface_state *surface_state = &xsurface->surface->current;
+	struct wlr_surface_state *state = &xsurface->surface->current;
 
 	if (view->swayc->instruction) {
 		transaction_notify_view_ready_by_size(view,
-				surface_state->width, surface_state->height);
-	} else if (container_is_floating(view->swayc)) {
-		view_update_size(view, surface_state->width, surface_state->height);
+				state->width, state->height);
+	} else if ((state->width != view->width || state->height != view->height) &&
+				container_is_floating(view->swayc)) {
+		// eg. The Firefox "Save As" dialog when downloading a file
+		// It maps at a small size then changes afterwards.
+		view->width = state->width;
+		view->height = state->height;
+		container_set_geometry_from_floating_view(view->swayc);
+		transaction_commit_dirty();
```

Comment:
Hmm. So the client updated its size, and `transaction_commit_dirty` will send a new configure event with the same size?

Reply:
Not quite. `transaction_commit_dirty` sees that the pending size matches the desired size, so it'll won't send a configure. The transaction will be applied immediately.

Reply:
Well `current` isn't updated, is it?

Reply:
It is. When the transaction is applied it copies the pending state to `current`.

Reply:
I don't get it. `should_configure` has this check:

```c
	if (con->current.view_width == instruction->state.view_width &&
			con->current.view_height == instruction->state.view_height) {
```

When `transaction_commit_dirty` is called, `current` still has the old state (the transaction isn't applied yet). So it still sends a configure event right?

Reply:
Yeah, `should_configure` correctly returns false, but the instruction is already added to the transaction (added in `transaction_add_container`). An instruction is not a configure. An instruction may or may not need to send a configure. If it helps, you could think of an instruction as a container state that needs to be applied when the transaction is ready.

Reply:
>Yeah, should_configure correctly returns false

No, `should_configure` returns `true`because the pending state (set here) mismatches the current state (set when the transaction is applied).

Reply:
Yeah, there's something wrong there. Let me mull over this.

Reply:
This is ready for review again.